store/sqlstore/reaction_store.go

```diff
@@ -26,7 +26,14 @@ func (s *SqlReactionStore) Save(reaction *model.Reaction) (*model.Reaction, erro
 	if err := reaction.IsValid(); err != nil {
 		return nil, err
 	}
-
+	if reaction.ChannelId == "" {
+		// get channelId before beginning transaction, if not already populated
+		chIds, err := s.Post().GetPostsByIds([]string{reaction.PostId})
```

Comment:
This does lots of extra things. I'd suggest to begin the transaction earlier, and just write a raw SQL query `select channelId from posts where id=?` to directly get the channel id.